### PR TITLE
Twitch-Chat-Downloader: init with dependencies

### PR DESCRIPTION
**Pull request by strager**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @berdario @thanegill

### CODE REVIEW

**Review by FRidh — CHANGES_REQUESTED** [review]
I can think of absolutely *no* good reason why a package would want to import from `pipenv`. 

I had a quick look at the `setup.py` and what is done there is just ridiculous. They use `pipenv ` to create an env, an then use `pipenv` to convert the locked dependencies for `setuptools`.  I get why they do that as it simplifies some of their work but its not what should be done in a `setup.py`. That whole part needs to be patched out before it can go in Nixpkgs.

Pipenv should not be module but remain an application.

File: pkgs/tools/networking/twitch-chat-downloader/default.nix
Comment (strager) [reply pkgs/tools/networking/twitch-chat-downloader/default.nix]: Thanks for pointing this out. For some reason, I assumed `-e` was not in effect.

File: pkgs/development/python-modules/rx/3.0.nix
Comment (strager) [reply pkgs/development/python-modules/rx/3.0.nix]: It looks like nothing uses Rx version 1.6 currently. I'll upgrade it in-place instead of adding a new version.

**Review by strager — COMMENTED** [review]
> I can think of absolutely _no_ good reason why a package would want to import from `pipenv`.
> 
> I had a quick look at the `setup.py` and what is done there is just ridiculous. They use `pipenv ` to create an env, an then use `pipenv` to convert the locked dependencies for `setuptools`. I get why they do that as it simplifies some of their work but its not what should be done in a `setup.py`. That whole part needs to be patched out before it can go in Nixpkgs.
> 
> Pipenv should not be module but remain an application.

Good points. Patching the pipenv-using packages was easy, so I'll do that instead of exposing pipenv as a library.

**Comment by strager** [comment]
GitHub says "Changes requested", but I made the changes. How do I tell GitHub that I made the requested changes and that I want this PR reviewed?

**Comment by jonringer** [comment]
not sure if it shows on your UI, but on main there's a arrow to the right which I can expand and dismiss.

File: maintainers/maintainer-list.nix
Comment (jonringer) [inline maintainers/maintainer-list.nix]: please add as separate commit
```
maintainers: add strager
```

File: pkgs/tools/networking/twitch-chat-downloader/default.nix
Comment (jonringer) [inline pkgs/tools/networking/twitch-chat-downloader/default.nix]:
```suggestion
    homepage = "https://github.com/PetterKraabol/Twitch-Chat-Downloader";
```

File: pkgs/top-level/python-packages.nix
Comment (jonringer) [inline pkgs/top-level/python-packages.nix]: please move into expression

File: pkgs/top-level/python-packages.nix
Comment (jonringer) [inline pkgs/top-level/python-packages.nix]: move into expression please
```suggestion
  twitch-python = disabledIf (! pythonAtLeast "3.7") (callPackage ../development/python-modules/twitch-python { });
```

File: pkgs/development/python-modules/twitch-python/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/twitch-python/default.nix]:
```suggestion
  version = "0.0.17";
  disabled = pythonOlder "3.6";
```

Comment (jonringer) [reply pkgs/development/python-modules/twitch-python/default.nix]: and the other file as well please

**Comment by jonringer** [comment]
Made an issue upstream https://github.com/PetterKraabol/Twitch-Python/issues/13

File: pkgs/top-level/python-packages.nix
Comment (strager) [reply pkgs/top-level/python-packages.nix]: Will do. I followed the convention set by other packages, such as docker-py, opencv, and meson, but I didn't see that many more packages use `disabled = ...;` instead.

**Comment by strager** [comment]
> not sure if it shows on your UI, but on main there's a arrow to the right which I can expand and dismiss.

I think I see the "v" arrow you're talking about. When I click it, a panel opens with the text "FRidh requested changes" and a "..." button. Clicking the "..." button shows a menu with one option: "See review". From there, I don't see any button labelled "dismiss".

File: pkgs/top-level/python-packages.nix
Comment (jonringer) [reply pkgs/top-level/python-packages.nix]: the disabledIf works anywhere, but the disabled attr is special to python packages. It's common to have interpreter specific disables, so it's more likely to be commonly used in python

**Comment by jonringer** [comment]
> > not sure if it shows on your UI, but on main there's a arrow to the right which I can expand and dismiss.
> 
> I think I see the "v" arrow you're talking about. When I click it, a panel opens with the text "FRidh requested changes" and a "..." button. Clicking the "..." button shows a menu with one option: "See review". From there, I don't see any button labelled "dismiss".

Selection_009

this is what mine looks like

File: pkgs/tools/networking/twitch-chat-downloader/default.nix
Comment (jonringer) [inline pkgs/tools/networking/twitch-chat-downloader/default.nix]:
```suggestion
  pname = "Twitch-Chat-Downloader";
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM, assuming everything has some tests

**Comment by strager** [comment]
> otherwise LGTM, assuming everything has some tests

I don't think any of these packages have significant tests of their own.

**Comment by strager** [comment]
> > > not sure if it shows on your UI, but on main there's a arrow to the right which I can expand and dismiss.
> > 
> > 
> > I think I see the "v" arrow you're talking about. When I click it, a panel opens with the text "FRidh requested changes" and a "..." button. Clicking the "..." button shows a menu with one option: "See review". From there, I don't see any button labelled "dismiss".
> 
> Selection_009
> 
> this is what mine looks like

I see only "See review":

Screen Shot 2020-02-19 at 23 33 07

**Comment by jonringer** [comment]
https://github.com/PetterKraabol/Twitch-Python/issues/13 was closed, if you want to wait until the next release, and update this, then I think we can have another go

**Comment by PetterKraabol** [comment]
Hi, I've removed the pipenv dependecy from setup.py in twitch-python and twitch-chat-downloader.

**Comment by strager** [comment]
Gross patch removed. (Thanks @jonringer and @PetterKraabol!) This PR is ready for review again.

**Comment by strager** [comment]
graphql-core's build seems to be broken. I'll investigate.

**Comment by FRidh** [comment]
@GrahamcOfBorg build twitch-chat-downloader

**Comment by marsam** [comment]
superseded by https://github.com/NixOS/nixpkgs/pull/112712

**Pull request closed without merging**
